fraud/testing.go

```diff
@@ -33,34 +35,33 @@ func (d *DummyService) Get(context.Context, ProofType) ([]Proof, error) {
 	return nil, nil
 }
 
-type mockStore struct {
-	headers    map[int64]*header.ExtendedHeader
-	headHeight int64
-}
+type H libheader.Header
+
+type mockStore mockstore.MockStore[H]
```

Comment:
Then we wouldn't be able to define methods on them; at least, that's the error I get when I call `headerMock.MockStore` instead of the custom mockStore

p.s. My knowledge of GO isn't deep, so if I sound too naive, I can skip this problem and take on another one 🙂

Reply:
Hey @richardgreg no worries! If you want to define additional methods on `MockStore`, you are welcome to add them in `libs/header/mocks/store.go` if they are necessary/useful :) So no need to alias the type here. We'd like to keep `MockStore` and all its methods within the scope of the `libs/header/mocks` package anyway.

Reply:
Hey @richardgreg are you still working on this? If not, we can take it over no problem!

Reply:
Hi @renaynay, I kept running into dead ends when attempting to fix this. It's evident I need to improve my GO skills. Yeah, you can take it over. 🙂